### PR TITLE
RDCC-3522: JUnit 4 → JUnit 5 Migration

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-3522

### Change description ###

JUnit 4 → JUnit 5 Migration

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/DeleteOrganisationIntTest.java [hunk 1]
```diff
@@ -19,12 +19,12 @@
 import static uk.gov.hmcts.reform.professionalapi.helper.OrganisationFixtures.someMinimalOrganisationRequest;
 
 @RunWith(SpringIntegrationSerenityRunner.class)
```
Comment (ishratt) [inline src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/DeleteOrganisationIntTest.java:21]: We need to remove all the @RunWith and use @ExtendWith instead

**Review by SujeethKumarAri — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/UpdateOrganisationTest.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
 import uk.gov.hmcts.reform.professionalapi.util.AuthorizationEnabledIntegrationTest;
 
 @RunWith(SpringIntegrationSerenityRunner.class)
```
Comment (ishratt) [inline src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/UpdateOrganisationTest.java:27]: replace @RunWith

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/docs/SwaggerPublisher.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
 @RunWith(SpringIntegrationSerenityRunner.class)
```
Comment (ishratt) [inline src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/docs/SwaggerPublisher.java:28]: replace @RunWith

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalExternalUserFunctionalTest.java [hunk 4]
```diff
@@ -490,7 +498,7 @@ public void findOrganisationPbaWithoutEmailByExternalUserShouldBeBadRequest() {
 
     @Test
     @ToggleEnable(mapKey = "OrganisationExternalController.deletePaymentAccountsOfOrganisation", withFeature = false)
```
Comment (ishratt) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalExternalUserFunctionalTest.java:500]: do all the tests with `@ToggleEnable` having `withFeature = false` pass without adding the `@ExtendWith(FeatureToggleConditionExtension.class)` annotation?

Comment (AdilOoze) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/ProfessionalExternalUserFunctionalTest.java:500]: Addressed here: https://github.com/hmcts/rd-professional-api/pull/1122/commits/711ccd95c9dc836ae3880e50cc47aa46ddf8ab1b

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/docs/SwaggerPublisher.java (hunk 3 above)
Comment (AdilOoze) [reply src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/docs/SwaggerPublisher.java:28]: Addressed here: https://github.com/hmcts/rd-professional-api/pull/1122/commits/711ccd95c9dc836ae3880e50cc47aa46ddf8ab1b

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/UpdateOrganisationTest.java (hunk 2 above)
Comment (AdilOoze) [reply src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/UpdateOrganisationTest.java]: Addressed here: https://github.com/hmcts/rd-professional-api/pull/1122/commits/711ccd95c9dc836ae3880e50cc47aa46ddf8ab1b

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/DeleteOrganisationIntTest.java (hunk 1 above)
Comment (AdilOoze) [reply src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/DeleteOrganisationIntTest.java]: Addressed here: https://github.com/hmcts/rd-professional-api/pull/1122/commits/711ccd95c9dc836ae3880e50cc47aa46ddf8ab1b

**Review by SujeethKumarAri — DISMISSED** [review]

**Review by SujeethKumarAri — DISMISSED** [review]

**Review by ishratt — DISMISSED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/util/FeatureToggleConditionExtension.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                    Optional<ToggleEnable> toggleEnable = findAnnotation(element, ToggleEnable.class);
+
+                    return toggleEnable.map(toggle -> {
+                        featureToggleService.mapServiceToFlag();
+                        flagName = featureToggleService.getLaunchDarklyMap().get(toggle.mapKey());
+
+                        final boolean isFlagEnabled =
+                                featureToggleService
+                                        .isFlagEnabled("rd_user_profile_api", flagName);
+
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/util/FeatureToggleConditionExtension.java:46]: can you check the service name ?  .isFlagEnabled("rd_user_profile_api", flagName);

Comment (AdilOoze) [reply src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/util/FeatureToggleConditionExtension.java:46]: Good catch - Addressed here: https://github.com/hmcts/rd-professional-api/pull/1122/commits/2402aca945bf5e65dcbaaa0f0784c154e46c21fd 👍🏼

**Review by SujeethKumarAri — APPROVED** [review]

**Review by rajkatla-hmcts — APPROVED** [review]

**Pull request merged**